wandb/sdk/wandb_settings.py

```diff
@@ -165,56 +112,32 @@ def _get_program_relpath_from_gitrepo(
     if os.path.exists(full_path_to_program):
         relative_path = os.path.relpath(full_path_to_program, start=root)
         if "../" in relative_path:
-            if _logger:
-                _logger.warning("could not save program above cwd: %s" % program)
+            if _logger is not None:
+                _logger.warning(f"Could not save program above cwd: {program}")
             return None
         return relative_path
 
-    if _logger:
-        _logger.warning("could not find program at %s" % program)
+    if _logger is not None:
+        _logger.warning(f"Could not find program at {program}")
     return None
 
 
-# the setting exposed to users as `dir=` or `WANDB_DIR` is actually
-# the `root_dir`. We add the `__stage_dir__` to it to get the full
-# `wandb_dir`
-def get_wandb_dir(root_dir: str) -> str:
-    # We use the hidden version if it already exists, otherwise non-hidden.
-    if os.path.exists(os.path.join(root_dir, ".wandb")):
-        __stage_dir__ = ".wandb" + os.sep
-    else:
-        __stage_dir__ = "wandb" + os.sep
-
-    path = os.path.join(root_dir, __stage_dir__)
-    if not os.access(root_dir or ".", os.W_OK):
-        wandb.termwarn("Path %s wasn't writable, using system temp directory" % path)
-        path = os.path.join(tempfile.gettempdir(), __stage_dir__ or ("wandb" + os.sep))
-
-    return path
-
-
-def _str_as_bool(val: Union[str, bool]) -> Optional[bool]:
-    ret_val = None
-    if isinstance(val, bool):
-        return val
-    try:
-        ret_val = bool(strtobool(val))
-    except (AttributeError, ValueError):
-        pass
-    return ret_val
-
-
-def _redact_dict(
-    d: Dict[str, Any],
-    unsafe_keys: Union[Set[str], FrozenSet[str]] = frozenset({"api_key"}),
-    redact_str: str = "***REDACTED***",
-) -> Dict[str, Any]:
-    """Redact a dict of unsafe values specified by their key."""
-    if not d or unsafe_keys.isdisjoint(d):
-        return d
-    safe_dict = d.copy()
-    safe_dict.update({k: redact_str for k in unsafe_keys.intersection(d)})
-    return safe_dict
+@enum.unique
+class Source(enum.IntEnum):
+    OVERRIDE: int = 0
+    BASE: int = 1  # fixme: audit this
```

Comment:
Would love to know a little more about this comment ;)

Reply:
https://youtu.be/HQ47glxcxr0?t=13 ;)

It's a general comment that we should use these more or ditch - e.g. ENTITY or WORKSPACE never come up in the code, ORG only come up in tests, etc.